src/main/java/formflow/library/FileController.java

```diff
@@ -181,4 +213,117 @@ RedirectView delete(
       return new RedirectView("/error");
     }
   }
+
+  /**
+   * @param httpSession  - The current httpsession
+   * @param submissionId - The submissionId of the file to be downloaded
+   * @param fileId       - The UUID of the file to be downloaded.
+   * @return ON SUCCESS: A ResponseEntity with a status of ok, a header, the contentType, the content size, and a responseBody
+   * that includes the file.
+   * <p>ON FAILURE: A ResponseEntity with a status of forbidden-403 or not found- 404</p>
+   */
+  @GetMapping("/file-download/{submissionId}/{fileId}")
+  public ResponseEntity<StreamingResponseBody> downloadSingleFile(
+      HttpSession httpSession,
+      @PathVariable String submissionId,
+      @PathVariable String fileId
+  ) {
+
+    Optional<UserFile> maybeFile = uploadedFileRepositoryService.findById(UUID.fromString(fileId));
+    if (maybeFile.isEmpty()) {
+      log.error(String.format("File with id %s may have already been deleted", fileId));
```

Comment:
It will help us debug if this were to log more details  (submission id with the file id) and note that the user was trying to delete it, but we couldn't find the file.  Currently you are suggesting _what might have_ happened, but it would be easier to debug if we stated _what is happening_ ("Request to delete fileId (%d), but file not found in database.  (submission id %d)")